### PR TITLE
Removes WPPostContentViewProvider.

### LINKED ISSUE #5250: Remove protocol WPPostContentViewProvider.
Protocol `WPPostContentViewProvider` is basically a `Post` object in disguise.

Evaluate a way to remove it, and potentially replace it with a common ancestor between the relevant post classes.

### PR DESCRIPTION
**Pull request by diegoreymendez**
Resolves #5250 by removing the remaining usage of `WPPostContentViewProvider`.

### Details:

Removes any remaining usage of `WPPostContentViewProvider`.

### Known code issues:

There are a few explicit casts I had to add to keep things working (for example [here](https://github.com/wordpress-mobile/WordPress-iOS/pull/5392/files#diff-051f42702d0c9000183971077a4e2c66R420)).  These are just temporary fixes.

The reason these casts needed to be added is that `PostCardTableViewCell` is not referencing a `Post` object as it should, and is instead referencing an `AbstractPost` object.  This will be addressed in an upcoming PR.

I created an issue [here](https://github.com/wordpress-mobile/WordPress-iOS/issues/5396).

### Testing:

1. Launch the app.
2. Go to the list of pages for a blog.
3. Perform the following actions: create a page, edit a page, view a page, trash a page, restore a page.

Needs review: @jleandroperez

### CODE REVIEW

**Comment by diegoreymendez** [comment]
@jleandroperez - This is now ready for review.

File: WordPress/Classes/Models/AbstractPost.h [hunk 1]
```diff
@@ -1,12 +1,11 @@
 #import <Foundation/Foundation.h>
 #import <CoreData/CoreData.h>
 #import "BasePost.h"
-#import "WPPostContentViewProvider.h"
```
Comment (jleandroperez) [inline WordPress/Classes/Models/AbstractPost.h]: 👋

**Comment by jleandroperez** [comment]
All test cases verified. Looking good!

:shipit:

**Comment by diegoreymendez** [comment]
Thanks Jorge!

**Comment by jleandroperez** [comment]
Anytime Diego! 😄

**Pull request merged**